code/modules/economy/holopay.dm

```diff
@@ -0,0 +1,249 @@
+/// Max range at which the hologram can be projected before it deletes
+#define MAX_HOLO_RANGE 4
+/// Minimum forced fee for holopay stations. Registers as "pay what you want."
+#define MIN_FEE 0
+/// Maximum forced fee. It's unlikely for a user to encounter this type of money, much less pay it willingly.
+#define MAX_FEE 5000
+
+/obj/machinery/holopay
+	name = "holographic pay stand"
+	desc = "an unregistered pay stand"
+	icon = 'icons/obj/economy.dmi'
+	icon_state = "card_scanner"
+	alpha = 150
+	anchored = TRUE
+	layer = FLY_LAYER
+	/// Owner of the holopay
+	var/mob/living/owner
+	/// ID linked to the holopay
+	var/obj/item/card/id/linked_card
+	/// Owner's linked account to the holopay
+	var/datum/bank_account/linked_account
+	/// Replaces the "pay whatever" functionality with a set amount when non-zero.
+	var/force_fee = 0
+	/// List of logos available for customization - via font awesome 5
+	var/static/list/available_logos = list("angry", "ankh", "band-aid", "cannabis", "cat", "cocktail", "coins", "comments-dollar",
+	"cross", "cut", "donate", "dna", "flask", "glass-cheers", "glass-martini-alt", "hand-holding-usd", "heart", "heart-broken",
+	"hamburger", "hat-cowboy-side", "money-check-alt", "music", "pizza-slice", "prescription-bottle-alt", "radiation", "robot", "smile",
+	"tram", "trash")
+	/// The brand icon chosen by the user
+	var/shop_logo = "donate"
+
+/obj/machinery/holopay/New(turf/location, mob/living/user, obj/item/card/id/card, datum/bank_account/account)
+	loc = location
+	owner = user
+	linked_card = card
+	linked_account = account
```

Comment:
You need to clear these variables on destroy. I'm also pretty sure that the account can be derived from the id card, wonder if it wouldn't be nicer to use that. If you could get a signal for the id card changing owners I'd say use it for owner too

Reply:
In addition we have no guarantee that say, the card will not be qdel'd while the owner human is just standing still. I think weakrefs would be safer here